Sources/RInAppMessaging/CampaignTriggerAgent.swift

```diff
@@ -22,12 +25,16 @@ internal struct CampaignTriggerAgent: CampaignTriggerAgentType {
             validator.validate { (campaign, triggeredEvents) in
                 do {
                     try eventMatcher.removeSetOfMatchedEvents(triggeredEvents, for: campaign)
-                    dispatcher.addToQueue(campaignID: campaign.id)
+                    if campaign.isTooltip {
+                        tooltipDispatcher.setNeedsDisplay(tooltip: campaign)
+                    } else {
+                        campaignDispatcher.addToQueue(campaignID: campaign.id)
+                    }
                 } catch {
                     // Campaign is not ready to be displayed
```

Comment:
Is there a logger for errors in IAM? Is is relevant to log the error here?

Reply:
There is but this is a valid state - not an error. When `removeSetOfMatchedEvents` fails it means that the campaign still needs more events to be logged